lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystemFactory.java

```diff
@@ -106,20 +99,20 @@ public TrinoFileSystem create(ConnectorIdentity identity)
         return new AzureFileSystem(httpClient, tracingOptions, auth, endpoint, readBlockSize, writeBlockSize, maxWriteConcurrency, maxSingleUploadSize);
     }
 
-    public static HttpClient createAzureHttpClient(OkHttpClient okHttpClient, HttpClientOptions clientOptions)
+    public static HttpClient createAzureHttpClient(ConnectionProvider connectionProvider, HttpClientOptions clientOptions)
     {
         Integer poolSize = clientOptions.getMaximumConnectionPoolSize();
-        // By default, OkHttp uses a maximum idle connection count of 5.
         int maximumConnectionPoolSize = (poolSize != null && poolSize > 0) ? poolSize : 5;
+        clientOptions.setMaximumConnectionPoolSize(maximumConnectionPoolSize);
 
-        return new OkHttpAsyncHttpClientBuilder(okHttpClient)
+        return new NettyAsyncHttpClientBuilder()
                 .proxy(clientOptions.getProxyOptions())
                 .configuration(clientOptions.getConfiguration())
-                .connectionTimeout(clientOptions.getConnectTimeout())
+                .connectTimeout(clientOptions.getConnectTimeout())
                 .writeTimeout(clientOptions.getWriteTimeout())
                 .readTimeout(clientOptions.getReadTimeout())
-                .connectionPool(new ConnectionPool(maximumConnectionPoolSize,
-                        clientOptions.getConnectionIdleTimeout().toMillis(), TimeUnit.MILLISECONDS))
+                .connectionProvider(connectionProvider)
+                .eventLoopGroup(new NioEventLoopGroup(maximumConnectionPoolSize)) // default is 2 * availableProcessors
```

Comment:
why do you want more than default?

Reply:
It's not more than, it's the same. The value that is being passed is the same as `azure.max-http-requests` which defaults to 2 * availableProcessors. The same default is in the NioEventLoopGroup which means that the configuration is the same

Reply:
co drop the comment or change it. As it reads - "I am overriding the default which is ..."

Reply:
Yeah. I can drop the comment